### PR TITLE
Use Yarn Workspaces

### PR DESCRIPTION
**Pull request by jonnywyatt**
https://yarnpkg.com/lang/en/docs/workspaces/

Yarn will manage the installation / hoisting of dependencies, instead of Lerna.

Lerna will still handle publishing of packages.

Have to make the repo `private` in order to work with workspaces to prevent publishing top level...?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,9 +36,7 @@
     "prefmt": "prettier --write '+(storybook|.storybook.native|lib)/**/*.+(js|json)' '*.md' '.github/**/*.md'",
     "fmt": "lerna run fmt",
     "packages:publish": "lerna publish --conventional-commits --yes --concurrency=1 --exact",
-    "packages:publish-dry-run": "npm run packages:publish -- --skip-git --skip-npm",
-    "update-deps": "lerna bootstrap --hoist --npm-client=yarn --concurrency=1",
-    "postinstall": "npm run update-deps --silent"
+    "packages:publish-dry-run": "npm run packages:publish -- --skip-git --skip-npm"
```
Comment (jimhunty) [inline package.json]: Should this be `yarn packages:publish...` ?

File: package.json [hunk 2]
```diff
@@ -36,9 +36,7 @@
     "prefmt": "prettier --write '+(storybook|.storybook.native|lib)/**/*.+(js|json)' '*.md' '.github/**/*.md'",
     "fmt": "lerna run fmt",
     "packages:publish": "lerna publish --conventional-commits --yes --concurrency=1 --exact",
-    "packages:publish-dry-run": "npm run packages:publish -- --skip-git --skip-npm",
-    "update-deps": "lerna bootstrap --hoist --npm-client=yarn --concurrency=1",
```
Comment (jimhunty) [inline package.json]: Will these lines be missed?

File: package.json (hunk 1 above)
Comment (jonnywyatt) [reply package.json]: Yes I should probably replace all npm commands with yarn

**Comment by rogeriopvl** [comment]
I'm all for this change 👍

I'm just wondering if we can we completely replace lerna with yarn workspaces?

**Comment by jonnywyatt** [comment]
@rogeriopvl I don't think yarn workspaces can provide the publish function that lerna has?

**Comment by rogeriopvl** [comment]
@jonnywyatt yes you're correct. I just noticed that for publishing it has to be used alongside lerna.

File: package.json (hunk 2 above)
Comment (jimhunty) [reply package.json]: 👍

Comment (craigbilner) [reply package.json]: @jonnywyatt can we keep the dry-run which is useful to debug publishing or test stuff out?

**Comment by killanaca** [comment]
@jonnywyatt do we still need lerna for the `run` command or we can migrate that too to Yarn workspace?

File: lib/cli/package-json.js [hunk 3]
```diff
@@ -34,11 +34,11 @@ module.exports = variables => ({
   },
   homepage: "https://github.com/newsuk/times-components#readme",
   devDependencies: {
-    "@times-components/jest-configurator": "0.0.1",
+    "@times-components/jest-configurator": "0.0.8",
```
Comment (craigbilner) [inline lib/cli/package-json.js]: @jimhunty FYI this will struggle to stay in sync

Comment (jimhunty) [reply lib/cli/package-json.js]: Why is that?

Comment (craigbilner) [reply lib/cli/package-json.js]: @jimhunty `jest-configurator` will auto-bump but `package-json.js` is a flat file and will not get the same treatment

Comment (jimhunty) [reply lib/cli/package-json.js]: Ah of course, sounds like we need a better way of doing the CLI 😉

**Comment by jimhunty** [comment]
I can confirm everything works as expected. I cleared all node modules from root and each of the packages, yarn installed and ran the web, iOS and Android environments successfully.

Good to go in I reckon.

**Pull request merged**